cmds/boot/stboot/main.go

```diff
@@ -0,0 +1,300 @@
+	}
+	url.Path = path.Join(url.Path, stboot.BallName)
+	err = downloadFromHTTPS(url.String(), ballPath)
+	if err != nil {
+		log.Fatalf("Downloading bootball failed: %v", err)
+	}
+
+	ball, err := stboot.BootBallFromArchie(ballPath)
+	if err != nil {
+		log.Fatal("Cannot open bootball")
+	}
+
+	// Just choose the first Bootconfig for now
+	log.Printf("Pick the first boot configuration")
+	var index = 0
+	bc, err := ball.GetBootConfigByIndex(index)
+	if err != nil {
+		log.Fatalf("Cannot get boot configuration %d: %v", index, err)
+	}
+
+	if *doDebug {
+		str, _ := json.MarshalIndent(*bc, "", "  ")
+		log.Printf("Bootconfig: %s", str)
+	}
+
+	n, err := ball.VerifyBootconfigByName(bc.Name)
+	if err != nil {
+		log.Fatalf("Bootconfig %d seems to be not trustworthy: %v", index, err)
+	}
+	if n < vars.MinimalSignaturesMatch {
+		log.Fatalf("Did not found enough valid signatures. %d valid, %d required", n, vars.MinimalSignaturesMatch)
+	}
+	log.Printf("Bootconfig '%s' passed verification", bc.Name)
+	log.Print(check)
+
+	if *dryRun {
+		debug("Dryrun mode: will not boot")
+		return
+	}
+
+	log.Println("Starting up new kernel.")
+
+	if err := bc.Boot(); err != nil {
+		log.Printf("Failed to boot kernel %s: %v", bc.Kernel, err)
+	}
+	// if we reach this point, no boot configuration succeeded
+	log.Print("No boot configuration succeeded")
+
+	return
+}
+
+func configureStaticNetwork(vars stboot.HostVars) error {
+	log.Printf("Setup network configuration with IP: " + vars.HostIP)
+	addr, err := netlink.ParseAddr(vars.HostIP)
+	if err != nil {
+		return fmt.Errorf("Error parsing HostIP string to CIDR format address: %v", err)
+	}
+
+	iface, err := netlink.LinkByName(eth)
+	if err != nil {
+		return fmt.Errorf("Error retrieving interface %s: %v", eth, err)
+	}
+
+	if err = netlink.AddrAdd(iface, addr); err != nil {
+		return fmt.Errorf("Error adding address: %v", err)
+	}
+
+	if err = netlink.LinkSetUp(iface); err != nil {
+		return fmt.Errorf("Error bringing up interface %s: %v", eth, err)
+	}
+
+	gateway, err := netlink.ParseAddr(vars.DefaultGateway)
+	if err != nil {
+		return fmt.Errorf("Error parsing GatewayIP string to CIDR format address: %v", err)
+	}
+
+	r := &netlink.Route{LinkIndex: iface.Attrs().Index, Gw: gateway.IPNet.IP}
+	if err = netlink.RouteAdd(r); err != nil {
+		return fmt.Errorf("Error setting default gateway: %v", err)
+	}
+
+	return nil
+}
+
+func configureDHCPNetwork() error {
+	log.Printf("Trying to configure network configuration dynamically...")
+
+	link, err := dhclient.IfUp(eth)
+	if err != nil {
+		return err
+	}
+
+	var links []netlink.Link
+	links = append(links, link)
+
+	var level dhclient.LogLevel
+	if *doDebug {
+		level = 1
+	} else {
+		level = 0
+	}
+	config := dhclient.Config{
+		Timeout:  time.Second * 6,
+		Retries:  4,
+		LogLevel: level,
+	}
+
+	r := dhclient.SendRequests(context.TODO(), links, true, false, config)
+	for result := range r {
+		if result.Interface == link {
```

Comment:
this is always gonna be true, so you can remove this if

Reply:
fixed